src/test/kotlin/com/karumi/maxibonkata/KarumiHQsTest.kt

```diff
@@ -0,0 +1,107 @@
+package com.karumi.maxibonkata
+
+import io.kotlintest.matchers.shouldBe
+import io.kotlintest.properties.forAll
+import io.kotlintest.specs.ShouldSpec
+import com.karumi.maxibonkata.Generators.DeveloperGenerator
+import com.karumi.maxibonkata.Generators.HungryDeveloperGenerator
+import com.karumi.maxibonkata.Generators.NotSoHungryDeveloperGenerator
+import io.kotlintest.properties.Gen
+
+class KarumiHQsTest : ShouldSpec() {
+    init {
+        "KarumiHQs" {
+            should("start the day with 10 maxibons") {
+                val office = KarumiHQs()
+
+                office.maxibonsLeft shouldBe 10
+            }
+
+            should("always has more than two maxibons in the fridge") {
+                forAll(DeveloperGenerator(), { developer ->
+                    val office = KarumiHQs()
+
+                    office.openFridge(developer)
+
+                    office.maxibonsLeft > 2
+                })
+            }
+
+            should("buy 10 more maxibons if there are less than 3 in the fridge") {
+                forAll(HungryDeveloperGenerator(), { developer ->
+                    val office = KarumiHQs()
+                    val initialMaxibons = office.maxibonsLeft
+
+                    office.openFridge(developer)
+
+                    val expectedMaxibons = calculateMaxibonsLeft(initialMaxibons, developer)
+                    office.maxibonsLeft == expectedMaxibons
+                })
+            }
+
+            should("request 10 more maxibons using the chat if there are less than 3 in the fridge") {
+                forAll(HungryDeveloperGenerator(), { developer ->
+                    val chat = MockChat()
+                    val office = KarumiHQs(chat)
+
+                    office.openFridge(developer)
+
+                    chat.messageSent == "Hi guys, I'm ${developer.name}. We need more maxibons!"
+                })
+            }
+
+            should("never request more maxibons to the team using the chat if there are more than 2 in the fridge") {
+                forAll(NotSoHungryDeveloperGenerator(), { developer ->
+                    val chat = MockChat()
+                    val office = KarumiHQs(chat)
+                    office.openFridge(developer)
+
+                    chat.messageSent == null
+                })
+            }
+
+            should("always has more than two maxibons in the fridge even if some karumies grab maxibons in group") {
+                forAll(Gen.list(DeveloperGenerator()), { developers ->
+                    val office = KarumiHQs()
+
+                    office.openFridge(developers)
+
+                    office.maxibonsLeft > 2
+                })
+            }
+
+            should("buy 10 more maxibons if there are less than 2 in the fridge when grabbing maxibons in group") {
+                forAll(Gen.list(DeveloperGenerator()), { developers ->
```

Comment:
You can use a list of Hungry developers to ensure more maxibons are bought. This way won't need to implement the logic in `calculateMaxibonsLeft`.

Reply:
Interesting approach, in this scenario it's obvious it will work and cover the same flows but I'm wondering if in this case we were just trying to see what happened with an arbitrary set of developers (hungry or not).

Reply:
As a general rule, I'd go for two different test to check those two scenarios and avoid the logic in the test. But I have no strong opinion. Especially here, where the scenario is perfectly explained.

Reply:
The ``calculateMaxibonsLeft`` method here is added on purpose to show the training attendants how not to write a test. We've copy pasted a piece of the production code just because we wanted to use property-based when we should not use any property here and just write a unit test. I wrote a solution for this kata using ES6 [here](https://github.com/pedrovgs/JavaScriptKatas/blob/master/test/maxibon/karumiHQs.spec.js) and I replaced this property with a table of test scenarios.